lib/sles4sap_publiccloud.pm

```diff
@@ -1093,4 +1094,36 @@ sub wait_for_zypper {
     die "Zypper is still locked after $max_retries retries, aborting (rc: 7)" if $retry >= $max_retries;
 }
 
+=head2 wait_for_idle
+
+    The function wraps the `cs_wait_for_idle` command, and restarts in case of timeout (once, this
+    time fatal) after displaying cluster information.
+
+=over 1
+
+=item B<$timeout> - The timeout (in seconds) for the command.
+
+=back
+=cut
+
+sub wait_for_idle {
+    my ($self, %args) = @_;
+    my $timeout = $args{timeout} // 120;
+
+    my $rc = $self->run_cmd(cmd => 'cs_wait_for_idle --sleep 5', timeout => $timeout, rc_only => 1, proceed_on_failure => 1);
+    if ($rc == 124) {
+        record_info("cs_wait_for_idle", "cs_wait_for_idle returned $rc, gathering info and retrying");
```

Comment:
```
cs_wait_for_idle returned $rc, gathering info and retrying
```
I think this message could be misleading. 124 does not come from `cs_wait_for_idle` but from bash `timeout` used to wrap call to it. And it was pretty confising point when looking at the log as the `timeout` wrapper things happens deeply in the perl code.

Maybe we can change it to 
```
"Timeout after $args{timeout} as executing cs_wait_for_idle returned. Gathering info and retrying"
```